gamemodes/terrortown/gamemode/server/sv_roleselection.lua

```diff
@@ -578,11 +584,20 @@ local function SelectForcedRoles(plys, selectableRoles)
 
 			selectedPlys[ply] = true
 		end
+
+		-- assigning the amount of forced players per role
+		selectedForcedRoles[subrole] = curCount
+
+		-- now assign amount of forced players per baserole if this is only a subrole
+		local baserole = roles.GetByIndex(subrole).baserole
+		if baserole and baserole ~= subrole then
```

Comment:
```suggestion
		if not roles.GetByIndex(subrole):IsBaseRole() then
```
ref. https://github.com/TTT-2/TTT2/blob/953f060584d0ad6975b3a2bce4b85ae5ad894e11/lua/terrortown/entities/roles/ttt_role_base/shared.lua#L89

Reply:
The function is nice, but is it faster? Because I still need the baserole index at line 594, though.

Reply:
Not directly faster, but even not a big difference while it's better to read. I suggests storing `roles.GetByIndex(subrole)` in another var so you don't need to loop through every role again. And later, you can use `roleData:GetBaseRole()` to get the connected baserole while not accessing .baserole directly but using the recommended functions.